airflow/api_fastapi/core_api/routes/public/backfills.py

```diff
@@ -206,3 +209,30 @@ def create_backfill(
             status_code=status.HTTP_409_CONFLICT,
             detail=f"There is already a running backfill for dag {backfill_request.dag_id}",
         )
+
+
+@backfills_router.post(
+    path="/dry_run",
+    responses=create_openapi_http_exception_doc(
+        [
+            status.HTTP_404_NOT_FOUND,
+            status.HTTP_409_CONFLICT,
+        ]
+    ),
+)
+def create_backfill_dry_run(
+    body: BackfillPostBody,
+) -> DryRunBackfillCollectionResponse:
+    from_date = timezone.coerce_datetime(body.from_date)
+    to_date = timezone.coerce_datetime(body.to_date)
+
+    backfills_dry_run = _do_dry_run(
+        dag_id=body.dag_id,
+        from_date=from_date,
+        to_date=to_date,
+        reverse=body.run_backwards,
+        reprocess_behavior=body.reprocess_behavior,
```

Comment:
and pass it there.

Usually 1 endpoint = 1 session. Opened at the beginning and committed on exit. But that's not blocking, just a suggestion to be consistent with other endpoints.